src/assets/toolkit/scripts/lib/component/slideshow.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+import {u} from 'umbrellajs';
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCountElement = element.querySelector('.js-Slideshow-current'),
+    totalCountElement = element.querySelector('.js-Slideshow-total'),
+    classIsVisible = 'is-visible',
+    classWasVisible = 'was-visible',
+    classIsSlidingForward = 'is-sliding-forward',
+    classIsSlidingBack = 'is-sliding-back',
+    classIsForward = 'is-forward',
+    classIsBack = 'is-back'
+  } = {}) {
+
+    Object.assign(this, {
+      slides,
+      slideHolder,
+      nextTrigger,
+      prevTrigger,
+      currentCountElement,
+      totalCountElement,
+      classIsVisible,
+      classWasVisible,
+      classIsForward,
+      classIsBack,
+      classIsSlidingForward,
+      classIsSlidingBack
+    });
+
+    this.counter = 0;
+    u(this.totalCountElement).html(this.numSlides);
+    u(this.nextTrigger).handle('click', () => this.next());
+    u(this.prevTrigger).handle('click', () => this.prev());
+  }
+
+  get numSlides() {
+    return this.slides.length;
+  }
+
+  slide(direction) {
+
+    // Get the index of the current item
+    if (this.counter > 0) {
+      var slideToShow = (this.counter % this.numSlides);
```

Comment:
Could we move the `slideToShow` variable before the `if`/`else` statement? Something like:

```js
slide(direction) {
  var slideToShow;

  if (this.counter > 0) {
    slideToShow = (this.counter % this.numSlides);
  } else {
    slideToShow = // else code here...
  }
  // rest of code...
}
```